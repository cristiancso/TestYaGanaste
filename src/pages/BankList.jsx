import { useFetch } from "../hooks/useFetch";

const BankList = () => {
  let bankList = null;
  if (localStorage.getItem('bankList')) {
    bankList = JSON.parse(localStorage.getItem('bankList'));
  }
  else {
    const { data } = useFetch('https://jsonplaceholder.typicode.com/users');
    if (!data) { return null; }
    bankList = data;
    localStorage.setItem('bankList', JSON.stringify(bankList));
  }

  return <div className="container">
    <h4>Bank List:</h4>
    <ul>
      {
        bankList.map((bank) => (
          <li key={bank.id}>{bank.name}</li>
        ))
      }
    </ul>
  </div>
}

export default BankList;