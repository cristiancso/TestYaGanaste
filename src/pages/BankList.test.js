import { render } from '@testing-library/react';
import BankList from './BankList';

describe('BankList', () => {
  test('renders without crashing', () => {
    const { container } = render(<BankList />)

    expect(container).toMatchSnapshot();
  });
});