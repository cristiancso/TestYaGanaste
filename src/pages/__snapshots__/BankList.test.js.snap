// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BankList renders without crashing 1`] = `<div />`;
